src/cloud-code/httpRequest.js

```diff
@@ -45,11 +57,11 @@ module.exports = function(options) {
     var httpResponse = {};
     httpResponse.status = response.statusCode;
     httpResponse.headers = response.headers;
-    httpResponse.buffer = new Buffer(response.body);
+    httpResponse.buffer = response.body;
     httpResponse.cookies = response.headers["set-cookie"];
-    httpResponse.text = response.body;
+    httpResponse.text = response.body.toString('utf-8');
```

Comment:
Ok, let me ask it a different way - can we not do this? Or rather, have text be a lazy evaluated function that is only converting a body into string when needed?

The piece that I am worried about is when we download an image from the server - will this try to convert it into string at all times and load it?